tests/test_ledger.py

```diff
@@ -1,445 +1,147 @@
-    assert len(spit_txn) == len(txn) + 2, "Expecting two new lines when transaction is split"
-    assert sum(spit_txn["account"] == transitory_account) == 2, (
-        "Expecting two transactions on transitory account"
-    )
-    assert all(spit_txn.groupby("id")["base_currency_amount"].sum() == 0), (
-        "Expecting split transactions to be balanced"
-    )
-    assert spit_txn.query("account == @transitory_account")["base_currency_amount"].sum() == 0, (
-        "Expecting transitory account to be balanced"
-    )
-
-
-def test_split_several_multi_currency_transactions():
-    cashctrl = ExtendedCashCtrlLedger(transitory_account = 19993)
-    transitory_account = 19993
-    txn = cashctrl.standardize_ledger(LEDGER_ENTRIES.query("id.isin([15, 16])"))
-    spit_txn = cashctrl.split_multi_currency_transactions(
-        txn, transitory_account=transitory_account
-    )
-    is_base_currency = spit_txn["currency"] == cashctrl.base_currency
-    spit_txn.loc[is_base_currency, "base_currency_amount"] = spit_txn.loc[
-        is_base_currency, "amount"
-    ]
-    id_currency_pairs = (txn["id"] + txn["currency"]).nunique()
-    assert len(spit_txn) == len(txn) + id_currency_pairs, (
-        "Expecting one new line per currency and transaction"
-    )
-    assert sum(spit_txn["account"] == transitory_account) == id_currency_pairs, (
-        "Expecting one transaction on transitory account per id and currency"
-    )
-    assert all(spit_txn.groupby("id")["base_currency_amount"].sum() == 0), (
-        "Expecting split transactions to be balanced"
-    )
-    assert spit_txn.query("account == @transitory_account")["base_currency_amount"].sum() == 0, (
-        "Expecting transitory account to be balanced"
-    )
-
-
-def test_mirror_ledger(set_up_vat_and_account):
-    cashctrl = ExtendedCashCtrlLedger(transitory_account = 19999)
-
-    # Mirror with one single and one collective transaction
-    target = LEDGER_ENTRIES.query("id in [1, 2]")
-    cashctrl.mirror_ledger(target=target, delete=True)
-    expected = cashctrl.standardize_ledger(target)
-    mirrored = cashctrl.ledger()
-    assert txn_to_str(mirrored) == txn_to_str(expected)
-
-    # Mirror with duplicate transactions and delete=False
-    target = pd.concat(
-        [
-            LEDGER_ENTRIES.query("id == 1"),
-            LEDGER_ENTRIES.query("id == 1").assign(id=5),
-            LEDGER_ENTRIES.query("id == 2").assign(id=6),
-            LEDGER_ENTRIES.query("id == 2"),
+    def test_add_ledger_entry(self, ledger, ledger_id):
+        ledger.restore(accounts=self.ACCOUNTS, vat_codes=self.VAT_CODES)
+        target = self.LEDGER_ENTRIES.query("id == @ledger_id")
+        id = ledger.add_ledger_entry(target)
+        remote = ledger.ledger()
+        created = remote.loc[remote["id"] == str(id)]
+        expected = ledger.standardize_ledger(target)
+        assert_frame_equal(
+            created, expected, ignore_index=True, ignore_columns=["id"], check_exact=True
+        )
```

Comment:
Wo should refer to the parent class method to test the addition of an account, rather than defining a specific method here.